include/fiction/algorithms/physical_design/hexagonalization.hpp

```diff
@@ -0,0 +1,200 @@
+                        const auto signal_b = lyt.incoming_data_flow(old_coord)[1];
+
+                        hex_layout.create_nor(hex_layout.make_signal(hex_layout.get_node(
+                                                  to_hex<Lyt>(signal_a.x, signal_a.y, layout_height, signal_a.z))),
+                                              hex_layout.make_signal(hex_layout.get_node(
+                                                  to_hex<Lyt>(signal_b.x, signal_b.y, layout_height, signal_b.z))),
+                                              hex);
+                    }
+                    else if (lyt.is_xor(node))
+                    {
+                        const auto signal_a = lyt.incoming_data_flow(old_coord)[0];
+                        const auto signal_b = lyt.incoming_data_flow(old_coord)[1];
+
+                        hex_layout.create_xor(hex_layout.make_signal(hex_layout.get_node(
+                                                  to_hex<Lyt>(signal_a.x, signal_a.y, layout_height, signal_a.z))),
+                                              hex_layout.make_signal(hex_layout.get_node(
+                                                  to_hex<Lyt>(signal_b.x, signal_b.y, layout_height, signal_b.z))),
+                                              hex);
+                    }
+                    else if (lyt.is_xnor(node))
+                    {
+                        const auto signal_a = lyt.incoming_data_flow(old_coord)[0];
+                        const auto signal_b = lyt.incoming_data_flow(old_coord)[1];
+
+                        hex_layout.create_xnor(hex_layout.make_signal(hex_layout.get_node(
+                                                   to_hex<Lyt>(signal_a.x, signal_a.y, layout_height, signal_a.z))),
+                                               hex_layout.make_signal(hex_layout.get_node(
+                                                   to_hex<Lyt>(signal_b.x, signal_b.y, layout_height, signal_b.z))),
+                                               hex);
+                    }
+                    else if (lyt.is_function(node))
+                    {
+                        const auto signal_a = lyt.incoming_data_flow(old_coord)[0];
+                        const auto signal_b = lyt.incoming_data_flow(old_coord)[1];
+
+                        const auto node_fun = lyt.node_function(node);
+
+                        hex_layout.create_node({hex_layout.make_signal(hex_layout.get_node(
+                                                    to_hex<Lyt>(signal_a.x, signal_a.y, layout_height, signal_a.z))),
+                                                hex_layout.make_signal(hex_layout.get_node(
+                                                    to_hex<Lyt>(signal_b.x, signal_b.y, layout_height, signal_b.z)))},
+                                               node_fun, hex);
+                    }
+                }
+            }
+        }
+    }
+    return hex_layout;
```

Comment:
Before returning the layout, you might want to make sure that all names and signal names have been preserved. See `utils/name_utils.hpp`.

Reply:
This comment doesn't seem to be resolved anymore.